src/services/piscina/wrapper.ts

```diff
@@ -0,0 +1,106 @@
+import Piscina from 'piscina';
+import path from 'path';
+import { ProcessorTransformationRequest, UserTransformationServiceResponse } from '../../types';
+import { FeatureFlags } from '../../middlewares/featureFlag';
+import logger from '../../logger';
+
+const isTest = process.env.NODE_ENV === 'test';
+
+type PiscinaConfig = {
+  maxThreads?: number;
+  minThreads?: number;
+  idleTimeout?: number;
+  maxQueue?: number;
+  concurrentTasksPerWorker?: number;
+};
+
+class PiscinaService {
+  private static instance: PiscinaService;
+
+  private piscina: Piscina | null = null;
+
+  private isInitialized = false;
+
+  private static getInstance(): PiscinaService {
+    if (!PiscinaService.instance) {
+      PiscinaService.instance = new PiscinaService();
+    }
+    return PiscinaService.instance;
+  }
+
+  private getPiscinaConfig(): PiscinaConfig {
+    const config: PiscinaConfig = {};
+
+    if (process.env.PISCINA_MAX_THREADS) {
+      config.maxThreads = parseInt(process.env.PISCINA_MAX_THREADS, 10);
+    }
+    if (process.env.PISCINA_MIN_THREADS) {
+      config.minThreads = parseInt(process.env.PISCINA_MIN_THREADS, 10);
+    }
+    if (process.env.PISCINA_IDLE_TIMEOUT) {
+      config.idleTimeout = parseInt(process.env.PISCINA_IDLE_TIMEOUT, 10);
+    }
+    if (process.env.PISCINA_MAX_QUEUE) {
+      config.maxQueue = parseInt(process.env.PISCINA_MAX_QUEUE, 10);
+    }
+    if (process.env.PISCINA_CONCURRENT_TASKS_PER_WORKER) {
+      config.concurrentTasksPerWorker = parseInt(
+        process.env.PISCINA_CONCURRENT_TASKS_PER_WORKER,
+        10,
+      );
+    }
+
+    return config;
+  }
+
+  private getPiscinaInstance(): Piscina {
+    if (!this.isInitialized || !this.piscina) {
+      throw new Error('Piscina worker pool not initialized. Call initialize first.');
+    }
+    return this.piscina;
+  }
+
+  public static initialize(): void {
+    const service = PiscinaService.getInstance();
+
+    if (process.env.USE_PISCINA !== 'true') {
```

Comment:
Is this supposed to happen? Shouldn't the caller of the `PiscinaService` not instantiate it in the first place if `USE_PISCINA` is `false`?

Reply:
Made this change. Marking resolved